src/hotspot/os/linux/os_linux.cpp

```diff
@@ -3957,14 +3957,15 @@ bool os::Linux::commit_memory_special(size_t bytes,
 
 char* os::Linux::reserve_memory_special_huge_tlbfs(size_t bytes,
                                                    size_t alignment,
+                                                   size_t page_size,
                                                    char* req_addr,
                                                    bool exec) {
   assert(UseLargePages && UseHugeTLBFS, "only for Huge TLBFS large pages");
   assert(is_aligned(req_addr, alignment), "Must be");
-  assert(is_aligned(req_addr, os::large_page_size()), "Must be");
+  assert(is_aligned(req_addr, page_size), "Must be");
   assert(is_aligned(alignment, os::vm_allocation_granularity()), "Must be");
-  assert(is_power_of_2(os::large_page_size()), "Must be");
-  assert(bytes >= os::large_page_size(), "Shouldn't allocate large pages for small sizes");
+  assert(is_power_of_2(page_size), "Must be");
+  assert(bytes >= page_size, "Shouldn't allocate large pages for small sizes");
```

Comment:
We could replace these two asserts with _page_sizes.contains.

Reply:
.. Well, the first one at least.

Reply:
Was just about to say just that :) The second one should trigger on the case you mention, if we ever tried to use large pages for too small sizes.